utils/docker/Dockerfile.centos.8

```diff
@@ -206,10 +206,18 @@ COPY utils utils
 COPY src src
 USER root:root
 RUN chown -R daos_server.daos_server /home/daos/daos/
+
+ARG COMPILER=gcc
+
+RUN [ "$COMPILER" != "icc" ] || \
+  { cp utils/oneAPI.repo /etc/yum.repos.d && \
```

Comment:
Should we have the corresponding code to use our own cache/mirror of this repo the way we do for all of the others?

Should we fetch this more directly (i.e. directly from the source) so that any changes made upstream don't have to be noticed by somebody (as broken CI) and then a PR has to grind through CI before such potential breakage can be repaired?

Reply:
> Should we have the corresponding code to use our own cache/mirror of this repo the way we do for all of the others?

Possibly.  I'm working on the Leap 15 Dockerfile so that it could be cached, and only fetched once per week (per docker host) however.

> Should we fetch this more directly (i.e. directly from the source) so that any changes made upstream don't have to be noticed by somebody (as broken CI) and then a PR has to grind through CI before such potential breakage can be repaired?

fetch what more directly?  If you mean the repofile then I can't imagine it changes very often.

Reply:
> fetch what more directly? If you mean the repofile then I can't imagine it changes very often.

Yeah, the repo file.  Maybe it won't change very often.  Maybe never at all.  But even if only changes seldomly, every time it does change it will mean CI is broken and **somebody** will have to drop what they are doing to go fix it.  There are not enough *somebody*s around to knowingly add a potential time-bomb to CI.

Really though, even if there were enough *somebody*s, we still shouldn't add what could be a potential time bomb to CI.  It's just to expensive and disruptive.

Reply:
I don't think there is a "source", and the repo file is fairly compact so I'd assume that if it changed the source would change as well.

I updated this PR however to only work with leap, and there it does pull the repo file from suse directly so this should be a non-issue now.

Reply:
> I don't think there is a "source",

Where did you get the contents of [utils/oneAPI.repo](https://github.com/daos-stack/daos/pull/7270/commits/a1cda4fd116b2fb218824dc4298541c9f8c8f93a?authenticity_token=xZDvOo5l6dNzakiFu5gczq2BSy61yqBOuUaxzTYgFGIIPj6KeXaw1wD7d0eG03LtXn4LSm3yRPDybj7WLqoheQ%3D%3D&file-filters%5B%5D=.repo#diff-990a205e881d499cf1ded873053065a712fbf65ebec5a967dcd583819fdc8561) from then?

There used to be instructions on how to `curl` a `.repo` file to access the repo.  Those seem to have (unfortunately) been replaced by videos and whatnot on how to download installers, etc.  I thought you just downloaded that from that previously known location.

> and the repo file is fairly compact so I'd assume that if it changed the source would change as well.

Well, the location of the repo file is usually considered a "well known" location so should not move.  But since the repo file directs the tools to finding the repo content, it can change.  New .repo files can even be included in package updates, etc.
 
> I updated this PR however to only work with leap, and there it does pull the repo file from suse directly so this should be a non-issue now.

Those same `zyyper` commands have paralle `dnf config-manager` options to add repos by URL, FWIW.

Reply:
https://www.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-linux/top/installation/install-using-package-managers/yum-dnf-zypper.html#yum-dnf-zypper is the documentation I was using.

Reply:
Interesting.  They did at one time have a downloadable `.repo` file, which is pretty typical.  Shame to see they are moving away from the typical repo installation method.

Funny how they council creating a .repo file and only using that with yum/dnf and use repo-adding commands with Suse.  The repo file creation/move that they council for yum/dnf would work just as well for zypper and using tools to add repos works just as well with yum/dnf.  Just quite inconsistent.